### ISSUE TITLE
Analyzer 'Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames.CSharpSimplifyTypeNamesDiagnosticAnalyzer' threw an exception of type 'System.InvalidOperationException' 

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#25352]
This was reported to me by @JasonBock. We don't know what line of code he was on, but saw this in 15.6:

```
Analyzer 'Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames.CSharpSimplifyTypeNamesDiagnosticAnalyzer' threw the following exception:'Exception occurred with following context:

Compilation: EditorConfigGenerator.CoreSyntaxTree:M:\JasonBock\Personal\Code Projects\EditorConfigGenerator\EditorConfigGenerator.Core\Styles\CSharpStyleVarWhenTypeIsApparentStyle.csSyntaxNode: this.Data.Add [MemberAccessExpressionSyntax]@[822..835) (18,52)-(18,65)System.InvalidOperationException:
Unexpected value 'NamedType' of type 'Microsoft.CodeAnalysis.SymbolKind'   at
Microsoft.CodeAnalysis.QualifyMemberAccess.AbstractQualifyMemberAccessDiagnosticAnalyzer`1.GetApplicableOptionFromSymbolKind(SymbolKind symbolKind)   at
Microsoft.CodeAnalysis.Diagnostics.SimplifyTypeNames.SimplifyTypeNamesDiagnosticAnalyzerBase`1.GetRemoveQualificationDiagnosticDescriptor(SemanticModel model, SyntaxNode node, OptionSet optionSet, CancellationToken cancellationToken)   at
Microsoft.CodeAnalysis.Diagnostics.SimplifyTypeNames.SimplifyTypeNamesDiagnosticAnalyzerBase`1.TrySimplifyTypeNameExpression(SemanticModel model, SyntaxNode node, AnalyzerOptions analyzerOptions, Diagnostic& diagnostic, CancellationToken cancellationToken)   at
Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames.CSharpSimplifyTypeNamesDiagnosticAnalyzer.<>c__DisplayClass2_0.<AnalyzeNode>g__descendIntoChildren|1(SyntaxNode n)   at
Microsoft.CodeAnalysis.SyntaxNode.ChildSyntaxListEnumeratorStack.PushChildren(SyntaxNode node, Func`2 descendIntoChildren)   at
Microsoft.CodeAnalysis.SyntaxNode.<DescendantNodesOnly>d__167.MoveNext()   at
Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames.CSharpSimplifyTypeNamesDiagnosticAnalyzer.AnalyzeNode(SyntaxNodeAnalysisContext context)   at
Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__43`1.<ExecuteSyntaxNodeAction>b__43_0(ValueTuple`2 data)   at
Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)-----'.
```

That lines up to this:

https://github.com/dotnet/roslyn/blob/bc16b0126f340014b39eaa0482f89d74463455dc/src/Features/Core/Portable/QualifyMemberAccess/AbstractQualifyMemberAccessDiagnosticAnalyzer.cs#L129

So somehow we tried qualifying the name of a named type and it didn't like.

**Comment by jasonmalinowski** [comment]
FYI: @jinujoseph reported during the MVP summit...might be new?

**Comment by sharwell** [comment]
I have no idea why, but `SemanticModel.GetSymbolInfo` on the expression `this.Data.Add` returned a named type symbol.

**Comment by CyrusNajmabadi** [comment]
No other reports or prism hits.  this code has been heavily refactored.  closing out.
